### PR TITLE
Internalize restart configuration in new schedule structure

### PR DESCRIPTION
**Pull request by joakim-hove**
This PR rewrites the restart configuration to the new Schedule parsing approach, this is the last step in the Schedule refactoring. The restart configuration is based on the keywords RTPSOL, RPTSCHED and RPTRST. It is insanely complex and not very logical; although I am guilty also of the previous implementation I hardly understood it - it seemed to have many insane twists just to get those f.... tests green[*]. Current PR also has many insane hoops in order to get the test green - hopefully slightly more comprehensible than the old one. But still a dark corner where children and those faint of heart should keep away.

[*]: Inverse programming programming completely in the blind - only to get the tests green!

### CODE REVIEW

**Comment by joakim-hove** [comment]
jenkins build this please

**Comment by joakim-hove** [comment]
jenkins build this please

**Comment by joakim-hove** [comment]
jenkins build this please

**Comment by joakim-hove** [comment]
jenkins build this please

**Comment by joakim-hove** [comment]
jenkins build this please

**Comment by joakim-hove** [comment]
jenkins build this please

**Comment by joakim-hove** [comment]
There are some small items left (a couple of warnings) - the --output-interval commandline option +++, but I consider this very close to merging. Happy for feedback.

**Comment by joakim-hove** [comment]
jenkins build this please

**Comment by joakim-hove** [comment]
jenkins build this opm-simulators=3136 please

**Comment by joakim-hove** [comment]
jenkins build this opm-simulators=3136 please

**Review by bska — APPROVED** [review]
This is a good rewrite.  Just one minor comment about some leftover debugging code,otherwise we should merge this.

File: src/opm/parser/eclipse/EclipseState/Schedule/KeywordHandlers.cpp [hunk 1]
```diff
@@ -790,8 +790,34 @@ namespace {
...
         this->snapshots.back().rpt_config.update( RPTConfig(handlerContext.keyword ));
+        auto rst_config = this->snapshots.back().rst_config();
+        rst_config.update(handlerContext.keyword, parseContext, errors);
+        if (handlerContext.currentStep == 3) {
+            const auto& write_rst = rst_config.write_rst_file;
+            if (write_rst.has_value())
+                printf("3: rst_config.write_rst_file: %d\n", write_rst.value());
+            else
+                printf("3: rst_config.write_rst_file: ??\n");
+        }
```
[the remark this answers is not in the document]
Comment (joakim-hove) [reply src/opm/parser/eclipse/EclipseState/Schedule/KeywordHandlers.cpp:803]: Thanks - yes that was obviously debugging leftovers

**Comment by joakim-hove** [comment]
jenkins build this opm-simulators=3136 please

**Pull request merged**